src/handlers/schedule-checkin.ts

```diff
@@ -63,67 +63,68 @@ async function handleInternal(event: AWSLambda.APIGatewayProxyEvent) {
     lastName: requestBody.data.last_name
   };
 
-  const firstLegDepartureDate = await findFirstLegDate(reservation);
+  const allDepartureDates = await findAllDepartureLegs(reservation);
 
-  if (!firstLegDepartureDate) {
+  if (!allDepartureDates) {
     const result: AWSLambda.APIGatewayProxyResult = {
       statusCode: HttpStatus.UNPROCESSABLE_ENTITY,
       headers: ResponseUtils.getStandardResponseHeaders(),
       body: JSON.stringify({ error: 'No future legs found', error_code: 'no_future_legs' })
     };
-
     return result;
   }
 
-  const checkinAvailableDateTime = Luxon.DateTime.fromJSDate(firstLegDepartureDate).minus({
-    hours: 24
-  });
-
-  console.debug('firstLegDepartureDate', firstLegDepartureDate);
+  const checkinAvailableDateTimes = allDepartureDates.map(date => date.minus({
+    hours:24
+  }))
+  
+  console.debug('allDepartureDates:', allDepartureDates.map(date => date.toJSDate()));
 
   // start checking in 5 minutes early (gives time for EventBridge trigger, Lambda cold start,
   // generating advanced checkin headers, etc.)
-  const invokeLambdaDateTime = checkinAvailableDateTime.minus({ minutes: 5 });
+  const responseBody: ResponseBody = {  
+    data: {
+      checkin_available_epoch: [],
+      checkin_boot_epoch: [],
+}};
```

Comment:
Minor formatting inconsistencies. The project has a prettier config. No big deal, not necessary to fix, but if you use VSCode and you'd like to use prettier, you can install it as an extension.